csharp_sdk_fix/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+# Note this is a separate file to include as early as possible in the root project when it is needed
+
+# This can go away when this cmake issue is resolved and we move to a version of cmake with that fix https://gitlab.kitware.com/cmake/cmake/-/issues/23513
+
+# So the SDK style projects we generate for C# .NET6 have a weird quirk in cmake (still present as of 3.26).
+# Cmake will generate the sln file such that ALL of these build as "Any CPU" and there is no way to avoid it
+# The symptom in Visual Studio is that the build solution will show many projects as "skipped" because we want to build x64
+# As a horrible hack, we can modify the sln file after Cmake is done.
+# How? Well, SO to the rescue: https://stackoverflow.com/questions/7091447/run-command-after-generation-step-in-cmake
+# "all" we have to do is recursively call cmake itself recursively (relying on any options passed to be cached)
+# then we can have "post-generate" step commands, which we will use to hack up the .sln and replace "Any CPU" with "x64"
```

Comment:
>.sln [](http://example.com/codeflow?start=83&length=4)

OCD: in the other places we refer to this as "sln", no dot.
Consider using the same all over the place.

Reply:
dot removed